### PR TITLE
feat: DocumentScanner returns DocumentScanningResult

### PR DESCRIPTION
**Pull request by fbernaly**
- return DocumentScanningResult
- refactor example app
- refactor Android native plugin: DocumentScanner.java and GoogleMlKitDocumentScannerPlugin.java

### CODE REVIEW

File: packages/google_mlkit_document_scanner/android/src/main/java/com/google_mlkit_document_scanner/GoogleMlKitDocumentScannerPlugin.java [hunk 1]
```diff
@@ -1,55 +1,19 @@
...
 import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
 import io.flutter.embedding.engine.plugins.FlutterPlugin;
-import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.MethodChannel;
-import io.flutter.plugin.common.PluginRegistry;
-
-public class GoogleMlKitDocumentScannerPlugin implements FlutterPlugin, ActivityAware, MethodChannel.MethodCallHandler,
-        PluginRegistry.ActivityResultListener {
 
+public class GoogleMlKitDocumentScannerPlugin implements FlutterPlugin, ActivityAware {
```
Comment (bensonarafat) [inline packages/google_mlkit_document_scanner/android/src/main/java/com/google_mlkit_document_scanner/GoogleMlKitDocumentScannerPlugin.java:10]: Okay, i got it. 
This is was what i miss. 
I was implementing ActivityAware in DocumentScanner class

File: packages/google_mlkit_document_scanner/android/src/main/java/com/google_mlkit_document_scanner/DocumentScanner.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    private static final String START = "vision#startDocumentScanner";
+    private static final String CLOSE = "vision#closeDocumentScanner";
+    private static final String TAG = "DocumentScanner";
+    private final Map<String, GmsDocumentScanner> instances = new HashMap<>();
+    private final ActivityPluginBinding binding;
+    private MethodChannel.Result pendingResult = null;
+    final private int START_DOCUMENT_ACTIVITY = 0x362738;
+
+    public DocumentScanner(ActivityPluginBinding binding) {
+        this.binding = binding;
```
Comment (bensonarafat) [inline packages/google_mlkit_document_scanner/android/src/main/java/com/google_mlkit_document_scanner/DocumentScanner.java:38]: Pass as a constructor. 
Perfect 👌

**Review by bensonarafat — APPROVED** [review]
@fbernaly 
Overall this more perfect. 

Thanks, i took note of the changes.

**Comment by fbernaly** [comment]
@bensonarafat : could you approve.

**Review by bensonarafat — APPROVED** [review]

**Review by bensonarafat — APPROVED** [review]

**Pull request merged**